test/recipes/02-test_errstr.t

```diff
@@ -132,3 +73,58 @@ sub split_error {
 
     return @erritems;
 }
+
+# Compares the first argument as string to each of the arguments 3 and on,
+# and returns an array of two elements:
+# 0:  True if the first argument matched any of the others, otherwise false
+# 1:  A string describing the test
+# The returned array can be used as the arguments to Test::More::ok()
+sub match_any {
+    my $first = shift;
+    my $desc = shift;
+    my @strings = @_;
+
+    if (scalar @strings > 1) {
+        $desc = "match '$first' ($desc) with one of ( '"
+            . join("', '", @strings) . "' )";
+    } else {
+        $desc = "match '$first' ($desc) with '$strings[0]'";
+    }
+
+    return ( scalar( grep { $first eq $_ } @strings ) > 0,
+             $desc );
+}
+
+sub match_opensslerr_reason {
+    my $errnum = shift;
+    my @strings = @_;
+
+    my $errnum_hex = sprintf "%x", $errnum;
+    my $reason =
+        ( run(app([ qw(openssl errstr), $errnum_hex ]), capture => 1) )[0];
+    $reason =~ s|\R$||;
+    $reason = ( split_error($reason) )[3];
+
+    return match_any($reason, $errnum, @strings);
+}
+
+sub match_syserr_reason {
+    my $errnum = shift;
+
+    my @strings = ();
+    # The POSIX reason string
+    push @strings, eval {
+          # Set $! to the error number...
+          local $! = $errnum;
+          # ... and $! will give you the error string back
+          $!
+    };
+    # The OpenSSL fallback string
+    push @strings, "reason($errnum)";
+
+    # We know that the system reasons are recorded by OpenSSL as an 'int'
+    # with an added set high bit, which always is POSIX::INT_MAX + 1.
+    my $syserr_flag = INT_MAX + 1;
+
+    return match_opensslerr_reason($syserr_flag | $errnum, $errnum, @strings);
```

Comment:
I think the second arg, `$errnum, should be left out here.

Reply:
Yup, you're right.  That was a left-over from previous coding.  Gone